lib/jumpy-view.coffee

```diff
@@ -40,58 +42,61 @@ class JumpyView extends View
-            @disposables.add atom.workspace.observeTextEditors (editor) =>
-                editorView = atom.views.getView(editor)
-                return if $(editorView).is ':not(:visible)'
-
+        mobx.useStrict(true)
+        @mobJumpy = mobx.observable(
+            # global
+            statusBarJumpy: @statusBarJumpy
+
+            # Labels -----------------
+            allPositions: mobx.asMap({})
+            setPositions: ->
+                mobx.action (key, value) =>
+                    @allPositions.set key, value
+            relevantPositions: -> # not necessarily reflected whith a 'no match'
+                mobx.computed =>
+                    mobx.asMap(@allPositions.entries().filter ([key]) =>
+                        key.startsWith @currentKeys)
+            clearAllPositions: mobx.action ->
+                @allPositions.clear()
+
+            # Input -----------------
+            currentKeys: ''
+            statusMessage: 'Jump Mode!'
+            acceptKey: ->
+                mobx.action (character) =>
+                    @currentKeys += character
+
+                    @statusBarJumpy?.classList.remove 'no-match'
+                    @setStatusMessage @currentKeys
+                    if !@isMatch.get()
+                        @statusBarJumpy?.classList.add 'no-match'
+                        @setStatusMessage 'No match!'
+            isMatch: ->
+                mobx.computed =>
+                    @allPositions.entries().find ([key]) =>
+                        key.startsWith @currentKeys
+            removeKey: ->
+                mobx.action ->
+                    @currentKeys = @currentKeys[...-1]
+            setStatusMessage: ->
+                mobx.action (message) ->
+                    @statusMessage = message
+            reset: mobx.action =>
+                @statusBarJumpy?.classList.remove 'no-match'
+                @mobJumpy.clearKeys()
                 for decoration in @decorations
-                    element = decoration.getProperties().item
-                    if element.textContent.indexOf(@firstChar) != 0
-                        element.classList.add 'irrelevant'
-        else if not @secondChar
-            @secondChar = character
-
-        if @secondChar
-            @jump() # Jump first.  Currently need the placement of the labels.
-            @clearJumpMode()
-
-    clearKeys: ->
-        @firstChar = null
-        @secondChar = null
-
-    reset: ->
-        @clearKeys()
-        for decoration in @decorations
-            decoration.getProperties().item.classList.remove 'irrelevant'
-        @statusBarJumpy?.classList.remove 'no-match'
-        @statusBarJumpyStatus?.innerHTML = 'Jump Mode!'
+                    decoration.getProperties().item
+                        .classList.remove 'irrelevant'
+            clearKeys: mobx.action ->
+                @currentKeys = ''
+                @statusMessage = 'Jump Mode!'
+                @statusBarJumpy?.innerHTML = ''
```

Comment:
Small thing but this line as well where you clear out the inner HTML.

My point being is that your UI (whatever it is React, Atom plugins etc) can behave independently of the state.